aurmr_tasks/src/aurmr_tasks/common/hri.py

```diff
@@ -0,0 +1,264 @@
+        marker.color.r = 1.0
+        marker.color.g = 0.0
+        marker.color.b = 0.0
+        marker.color.a = 1.0
+
+        marker.lifetime = rospy.rostime.Duration()
+
+        self.marker_publisher.publish(marker)
+    
+    def visualize_point_marker(self, point, frame_id):
+        marker2 = Marker()
+        # marker2.header.frame_id = self.camera_model.tfFrame()
+        marker2.header.frame_id = frame_id
+        marker2.header.stamp = rospy.rostime.Time.now()
+        marker2.ns = "basic_shapes"
+        marker2.id = 1
+        marker2.type = Marker.SPHERE
+        marker2.action = Marker.ADD
+
+        print(f"publishing {point}")
+        marker2.pose.position.x = point[0]
+        marker2.pose.position.y = point[1]
+        marker2.pose.position.z = point[2]
+        marker2.pose.orientation.x = 0.0;
+        marker2.pose.orientation.y = 0.0;
+        marker2.pose.orientation.z = 0.0;
+        marker2.pose.orientation.w = 1.0;
+
+        # marker2.points = [p1, p2]
+
+        # Set the scale of the marker -- 1x1x1 here means 1m on a side
+        marker2.scale.x = 0.05
+        marker2.scale.y = 0.05
+        marker2.scale.z = 0.05
+
+        # Set the color -- be sure to set alpha to something non-zero!
+        marker2.color.r = 0.0
+        marker2.color.g = 1.0
+        marker2.color.b = 0.0
+        marker2.color.a = 1.0
+
+        marker2.lifetime = rospy.rostime.Duration()
+
+        self.marker_publisher.publish(marker2)
+
+    def get_xyz(self, u, v):
+        ray = self.camera_model.projectPixelTo3dRay((u, v))
+        ray_z = [el/ray[2] for el in ray]
+
+        self.visualize_ray(ray, self.camera_model.tfFrame())
+
+        if self.use_depth:
+            # Get xyz using depth
+            depth_image = ros_numpy.numpify(self.ros_depth_image).astype(np.float32)
+            depth_z = depth_image[int(v), int(u)]
+            camera_point = np.array(ray_z) * (depth_z/1000)
+            return camera_point
+        else:
+            # Get xyz using pointcloud
+            # pc_baselink = self.tf_buffer.transform(self.ros_pointcloud, 'base_link')
+            pc = ros_numpy.point_cloud2.pointcloud2_to_xyz_array(self.ros_pointcloud, remove_nans=True)
+            p1 = np.array([0,0,0])
+            p0 = np.array(ray_z) * 2
+            closest_point = pc[np.argmin(np.linalg.norm(np.cross(p1-p0, p0-pc, axisb=1), axis=1)/np.linalg.norm(p1-p0))]
+            return closest_point
+
+
+    def execute(self, userdata):
+        if self.ros_pointcloud is None or self.ros_rgb_image is None or self.camera_model is None:
+            rospy.logwarn("Pointcloud and RGB image not ready for HRI regrasp")
+            return "continue"
+
+        if userdata['target_bin_id'] not in bin_bounds:
+            rospy.logwarn(f"No bin configuration found for {userdata['target_bin_id']}")
+            return "continue"
+
+        client = actionlib.SimpleActionClient('/aurmr/hri/retry_grasp', RetryGraspAction)
```

Comment:
`ActionClient` and other objects that create ROS Subscribers are expensive (I believe some of the resources/threads that service the message queues don't get cleaned up often), so it's common to make a single client in the constructor. Won't become a noticeable problem unless this state were used more a few dozen times in the same process.

Reply:
The reason why I instantiated it locally here is because the ActionServer in this case is flaky (it is implemented by our web app -- and could disconnect/reconnect if the user refreshes their page or closes their laptop to take a break).  So we connect with a short `wait_for_server` timeout to see if anyone is serving the action for this pick and a longer `wait_for_result` timeout once we've connected and sent them the goal.   Although, maybe the SimpleActionClient and/or rosbridge websocket server automatically handles those cases gracefully?  For now, I will merge this... but I will test more and maybe move this to the constructor if it doesn't cause issues when the server is intermittently available.